subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/Checkstyle.java

```diff
@@ -198,6 +200,9 @@ private void runWithProcessIsolation() {
             spec.getForkOptions().setMinHeapSize(minHeapSize.getOrNull());
             spec.getForkOptions().setMaxHeapSize(maxHeapSize.getOrNull());
             spec.getForkOptions().setExecutable(javaLauncher.get().getExecutablePath().getAsFile().getAbsolutePath());
+            if (getEnableExternalDtdLoad().isPresent()) {
+                spec.getForkOptions().getSystemProperties().put("checkstyle.enableExternalDtdLoad", getEnableExternalDtdLoad().get());
+            }
```

Comment:
I was trying to avoid setting the convention in the constructor, and also to avoid needlessly setting an extra system property set to false (which means the same as absence) when running the checkstyle process.

Are you sure we shouldn't leave the `@Optional` and use `getOrNull()` to match the other fork options (these do NOT create system props, but still)?